### PR TITLE
Remove PanZoomMesh in favour of individual meshes

### LINKED ISSUE #769: [Zoom] Give more zooming options
### Example use case 
With a plot like this:
image

I would like to be able to zoom on the y-axis while keeping the full x-axis. The current zoom only allows me to zoom on both axes at once.

### Implementation
We could imagine 
- Draw a zooming box with drag and drop ? This could interfere with the panning though.
- Zoom only on one axis with Alt/Ctrl+wheel ?
- Being able to zoom only on one axis by hovering on the desired axis and use the wheel

### PR DESCRIPTION
**Pull request by loichuder**
Fix #769

This time, `PanZoomMesh` is truly gone ! We now favour the use of the individual meshes:
- `PanMesh` to enable panning in vis bounds
- `ZoomMesh` to enable zooming in X and Y on the mouse position
- `XAxisZoomMesh` to enable zooming in X on the mouse position
- `YAxisZoomMesh` to enable zooming in Y on the mouse position

All meshes can be disabled and accept a `modifierKey` through props (with `Alt` the default for X and `Shift` the default for Y, the others have no default).

Stories and visualisations have been updated with the new components. Notably, zooming in X and/or Y is now allowed in `Line` (and its cousin `NxSpectrum`).

### CODE REVIEW

File: apps/storybook/src/VisCanvasInteraction.stories.tsx [hunk 1]
```diff
@@ -1,24 +1,52 @@
...
+    tooltipValue,
+    guides,
+    disablePan,
+    panKey,
+    disableZoom,
+    disableXZoom,
+    xZoomKey,
+    disableYZoom,
+    yZoomKey,
+  } = args;
```
Comment (axelboc) [inline apps/storybook/src/VisCanvasInteraction.stories.tsx:39]: Because of all the prop aliases, I feel like this template no longer documents the new meshes very well. Perhaps we should have one template per mesh? We can always store the canvas' axis configs into a shared object that we then destructure. Might also be worth splitting all the pan/zoom interactions into a separate file to stay clear of the tooltip? All of this can be left to a future PR.

File: packages/lib/src/vis/shared/XAxisZoomMesh.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  const onWheel = useZoomOnWheel(zoomVector, disabled);
+
+  return (
+    <mesh onWheel={onWheel}>
+      <meshBasicMaterial opacity={0} transparent />
+      <planeGeometry args={[width, height]} />
+    </mesh>
+  );
```
Comment (axelboc) [inline packages/lib/src/vis/shared/XAxisZoomMesh.tsx:36]: Can be left to another PR, but I feel like this could be abstracted to a `CanvasMesh` of sorts.

File: packages/lib/src/vis/shared/XAxisZoomMesh.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  modifierKey?: ModifierKey;
+}
+
+function XAxisZoomMesh(props: Props) {
+  const { disabled, modifierKey = 'Alt' } = props;
+
+  const { width, height } = useThree((state) => state.size);
+
+  const zoomVector = (sourceEvent: WheelEvent) => {
+    const factor = sourceEvent.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
```
Comment (axelboc) [inline packages/lib/src/vis/shared/XAxisZoomMesh.tsx:20]: Could this be computed in `useZoomOnWheel` and passed as an argument? `ZOOM_FACTOR` is the same everywhere, so there's no point to duplicate this logic for now, I reckon.

**Review by axelboc — APPROVED** [review]
:tada: :champagne: 

FYI, I've added a reference to the issue in the PR description.

File: apps/storybook/src/VisCanvasInteraction.stories.tsx (hunk 1 above)
Comment (loichuder) [reply apps/storybook/src/VisCanvasInteraction.stories.tsx:39]: No you are right, I might as well do it in this PR to properly document each mesh.

Comment (loichuder) [reply apps/storybook/src/VisCanvasInteraction.stories.tsx]: So I split into two files: one for the tooltip and one for the PanZoom meshes. In PanZoom, each story documents an individual mesh with controls that reflect its props and that act only on the relevant mesh. 

But all meshes are still present in each story as `Zoom` is needed for `Pan` to be used.

Tell me what you think.

File: packages/lib/src/vis/shared/XAxisZoomMesh.tsx (hunk 2 above)
Comment (loichuder) [reply packages/lib/src/vis/shared/XAxisZoomMesh.tsx]: Abstracted in `InteractionMesh`

File: packages/lib/src/vis/shared/XAxisZoomMesh.tsx (hunk 3 above)
Comment (loichuder) [reply packages/lib/src/vis/shared/XAxisZoomMesh.tsx]: I changed this to return booleans instead. The conversion in `zoomVector` is then done in the callback
